### PR TITLE
Add informative error message when aperture doesn't return a stream

### PR DESCRIPTION
**Pull request by thisRaptori**
For context, look at #149 

Didn't add tests for Preact and Inferno, since we're apparently not testing those libraries much yet and I couldn't quickly figure out how to do so... seems like that might be something worth fixing! 😬

### CODE REVIEW

File: base/react/configureComponent.ts [hunk 1]
```diff
@@ -134,6 +135,12 @@ const configureComponent = <Props, Effect, Context>(
 
     const sinkObservable = aperture(component, instance.props, instance.context)
 
+    if (!sinkObservable) {
+        throw new Error(
+            `Refract was unable to subscribe within your component (${componentName}). Make sure you return an observable from your aperture.`
```
Comment (thisRaptori) [inline base/react/configureComponent.ts]: Not sure if `within` is the right word here - the `withEffects` is _wrapping_ the `componentName` component; any alternative I could think of had much more words and was therefore _less_ clear than this... 😕 

Also kinda torn on whether it should say `observable` or not, considering `callbag`!

File: base/react/withEffects.ts [hunk 2]
```diff
@@ -64,7 +64,8 @@ export const withEffects = <Props, Effect, ChildProps = Props, Context = any>(
                 config.handler,
                 config.errorHandler,
                 config.mergeProps,
-                config.decorateProps !== false
+                config.decorateProps !== false,
+                BaseComponent.name
```
Comment (troch) [inline base/react/withEffects.ts]: It should be `BaseComponent.displayName || BaseComponent.name`. Same for Inferno and Preact.

File: base/react/configureComponent.ts (hunk 1 above)
Comment (troch) [reply base/react/configureComponent.ts]: `Your Refract aperture didn't return an observable entity in '${componentName}'`?

File: base/react/configureHook.ts [hunk 3]
```diff
@@ -74,6 +75,12 @@ export const configureHook = <Data, Effect>(
 
     const sinkObservable = aperture(component, data)
 
+    if (!sinkObservable) {
+        throw new Error(
+            `Your Refract aperture didn't return an observable entity in ${hookName}.`
```
Comment (troch) [inline base/react/configureHook.ts]: I would add `in hook '${hookName}'` to be more explicit if someone uses both hooks and HoCs

File: base/react/configureComponent.ts [hunk 4]
```diff
@@ -134,6 +135,12 @@ const configureComponent = <Props, Effect, Context>(
 
     const sinkObservable = aperture(component, instance.props, instance.context)
 
+    if (!sinkObservable) {
+        throw new Error(
+            `Your Refract aperture didn't return an observable entity in ${componentName}.`
```
Comment (troch) [inline base/react/configureComponent.ts]: I would add `in component '${componentName}'` to be more explicit if someone uses both hooks and HoCs (with quotes as well)

**Pull request merged**